apis/vshn/v1/vshn_nextcloud.go

```diff
@@ -0,0 +1,288 @@
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []VSHNNextcloud `json:"items,omitempty"`
+}
+
+// VSHNNextcloudSpec defines the desired state of a VSHNNextcloud.
+type VSHNNextcloudSpec struct {
+	// Parameters are the configurable fields of a VSHNNextcloud.
+	Parameters VSHNNextcloudParameters `json:"parameters,omitempty"`
+
+	// WriteConnectionSecretToRef references a secret to which the connection details will be written.
+	WriteConnectionSecretToRef v1.LocalObjectReference `json:"writeConnectionSecretToRef,omitempty"`
+}
+
+// VSHNNextcloudParameters are the configurable fields of a VSHNNextcloud.
+type VSHNNextcloudParameters struct {
+	// Service contains nextcloud DBaaS specific properties
+	Service VSHNNextcloudServiceSpec `json:"service,omitempty"`
+
+	// Size contains settings to control the sizing of a service.
+	Size VSHNSizeSpec `json:"size,omitempty"`
+
+	// Scheduling contains settings to control the scheduling of an instance.
+	Scheduling VSHNDBaaSSchedulingSpec `json:"scheduling,omitempty"`
+
+	// Backup contains settings to control how the instance should get backed up.
+	Backup K8upBackupSpec `json:"backup,omitempty"`
+
+	// Restore contains settings to control the restore of an instance.
+	Restore K8upRestoreSpec `json:"restore,omitempty"`
+
+	// Maintenance contains settings to control the maintenance of an instance.
+	Maintenance VSHNDBaaSMaintenanceScheduleSpec `json:"maintenance,omitempty"`
+
+	// Monitoring contains settings to control monitoring.
+	Monitoring VSHNMonitoring `json:"monitoring,omitempty"`
+
+	// Security defines the security of a service
+	Security Security `json:"security,omitempty"`
+
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=3
+
+	// Instances configures the number of Nextcloud instances for the cluster.
+	// Each instance contains one Nextcloud server.
+	Instances int `json:"instances,omitempty"`
+}
+
+// VSHNNextcloudserviceSpec contains nextcloud DBaaS specific properties
+type VSHNNextcloudServiceSpec struct {
+	// FQDN contains the FQDN which will be used for the ingress.
+	// If it's not set, no ingress will be deployed.
+	// This also enables strict hostname checking for this FQDN.
+	FQDN string `json:"fqdn,omitempty"`
+
+	// RelativePath on which Nextcloud will listen.
+	// +kubebuilder:default="/"
+	RelativePath string `json:"relativePath,omitempty"`
+
+	// +kubebuilder:default="29"
+
+	// Version contains supported version of nextcloud.
+	// Multiple versions are supported. The latest version 29 is the default version.
+	Version string `json:"version,omitempty"`
+
+	// +kubebuilder:validation:Enum="besteffort";"guaranteed"
+	// +kubebuilder:default="besteffort"
+
+	// ServiceLevel defines the service level of this service. Either Best Effort or Guaranteed Availability is allowed.
+	ServiceLevel VSHNDBaaSServiceLevel `json:"serviceLevel,omitempty"`
+
+	// // +kubebuilder:default=true"
+	//
```

Comment:
Please add a comment here that if this paramater is false, Nextcloud will use the default non managed sql lite database. Also add the fact that the PostgreSQL is fully managed.

Reply:
Added the comment